packages/react-compose/src/resolveSlotProps.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { OptionsResolverResult, defaultMappedProps } from './createOptionsResolver';
+
+export const NullRender = () => null;
+
+/**
+ * Helper utility which takes in a classes array from compose options, resolves functions,
+ * merges them into a final result, and distributes classnames to slotProps within the given
+ * resolver result object.
+ */
+export function resolveSlotProps(result: OptionsResolverResult): OptionsResolverResult {
+  const { state, slots, slotProps } = result;
+
+  Object.keys(slots).forEach((slotName: string) => {
+    const slot = slots[slotName];
+    let slotProp = state[slotName];
```

Comment:
Something to think about, it seems like the classnames could get passed through here and get resolved since each className must be associated with a particular slot. At least I think I'm reading that right.

Reply:
That's true, but there's more involved with resolving classnames done in the separate resolver (resolving functions with state) so it will be done separately. But yes perhaps we could combine the two loops. Trying to keep it clean for now, can t